Sources/AWSSDKSwiftCore/Message/AWSRequest.swift

```diff
@@ -100,23 +100,39 @@ public struct AWSRequest {
 
     /// Create HTTP Client request from AWSRequest
     func toHTTPRequest() -> AWSHTTPRequest {
-        return AWSHTTPRequest.init(url: url, method: HTTPMethod(rawValue: httpMethod), headers: getHttpHeaders(), body: body.asByteBuffer())
+        return AWSHTTPRequest.init(url: url, method: HTTPMethod(rawValue: httpMethod), headers: getHttpHeaders(), body: body.asPayload())
     }
 
     /// Create HTTP Client request with signed URL from AWSRequest
     func toHTTPRequestWithSignedURL(signer: AWSSigner) -> AWSHTTPRequest {
         let method = HTTPMethod(rawValue: httpMethod)
-        let bodyData = body.asByteBuffer()
-        let signedURL = signer.signURL(url: url, method: method, body: bodyData != nil ? .byteBuffer(bodyData!) : nil, date: Date(), expires: 86400)
-        return AWSHTTPRequest.init(url: signedURL, method: method, headers: getHttpHeaders(), body: bodyData)
+        let payload = self.body.asPayload()
+        let bodyDataForSigning: AWSSigner.BodyData?
+        if let body = payload?.asByteBuffer() {
+            bodyDataForSigning = .byteBuffer(body)
+        } else if case .stream = payload {
+            bodyDataForSigning = .unsignedPayload
+        } else {
+            bodyDataForSigning = nil
+        }
```

Comment:
use a switch?

Reply:
yeah not sure why that isnt a switch will sort

Reply:
done